### DISCUSSION TITLE
Is there a way to save all the tables inside an html page?

### DISCUSSION THREAD

**Discussion by aborruso** [saulpw/visidata#965] [Q&A] [answered]
Hi,
if I run

```
vd "https://www.regione.liguria.it/homepage/salute-e-sociale/homepage-coronavirus/bollettino-coronavirus/dati-25-4-2021.html"
```

I have 6 tables

![image](https://user-images.githubusercontent.com/30607/116153171-30b8b680-a6e7-11eb-9f66-711287a6cb45.png)

Is there a way to run a command in batch mode to save  all the table (in example as TSV)?
I know I can record a log file, but sometime this page changes and today I have 6 tables and tomorrow I will have 8

Thank you

**Comment by saulpw** [comment, marked as answer]
How about `gs` to select all the tables, and then `g Ctrl+S` to save all selected sheets?  You'd have to save to a format that supports multiple tables, or a directory, of course.

**Reply by aborruso** [reply]
Thank you very much. Really simple and powerful

**Reply by aborruso** [reply]
No I must be able to write a right log, because the input URL could change

**Reply by saulpw** [reply]
Do you mean that you want to make a cmdlog that you can replay even with a different url?  If so, you should just have to clear the 'sheet' column, as replay will use the top sheet for any command that doesn't provide a sheet name.

**Reply by aborruso** [reply]
It's a kind of magic

```
vd -y -b -p mm.vd input=https://www.regione.liguria.it/homepage/salute-e-sociale/homepage-coronavirus/bollettino-coronavirus/dati-21-4-2021.html
```

```
sheet	col	row	longname	input	keystrokes	comment
global		null_value	set-option			
			open-file	{input}	o	
			select-rows		gs	select all rows
			save-selected	./	g^S	save all selected sheets to given file or directory
```

**Reply by aborruso** [reply]
@saulpw is there some documentation page in which I can read about this use of `{input}` inside a log file, coupled with cli parameters?

Thank you

**Reply by aborruso** [reply]
@saulpw I have it ` field=value             replace "{field}" in cmdlog contents with value`

**Reply by saulpw** [reply]
Yes, I believe that's documented on the site somewhere.  If you found it, can you post a link?

**Reply by aborruso** [reply]
It's in the main doc
https://www.visidata.org/docs/v2.4/man/
